security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -471,3 +471,7 @@ func (ms *mockSecretStore) DeleteSecret(conID, resourceName string) {
 	}
 	ms.secrets.Delete(key)
 }
+
+func (ms *mockSecretStore) ShouldWaitForIngressGatewaySecret(connectionID, resourceName, token string) bool {
+	return false
```

Comment:
So this mocks out the new code instead of testing it.  It'd be nice to see a test for the new code.

Reply:
Looks like the test case arrived at the same time I added this comment.  Thanks for adding it

Reply:
The mock in this test is to test success cases, which means a SDS response is always sent out from the server, and the test verifies that response contains expected key/cert. 

I don't find an easy way to test the case when ShouldWaitForIngressGatewaySecret returns true. Because when ShouldWaitForIngressGatewaySecret return true, [stream.Recv()](https://github.com/istio/istio/blob/b43c5a1704164aa5d3a72f7bc67f0df6bdf9f28e/security/pkg/nodeagent/sds/sdsservice_test.go#L395) in the test does not return, the server just holds the stream and waits for secret to be ready. Without this change, the server sends an error message and that causes the gateway to send another SDS request immediately, which causes the memory issue.

Maybe I can add some e2e tests for the ingress sds in tests/e2e, or add a new test framework in nodeagent/ to support testing this scenario. Is it okay to let these tests in a separate PR?